src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor.cs

```diff
@@ -120,6 +123,23 @@ private void AddEnumConstantValue(INamedTypeSymbol enumType, object constantValu
             }
         }
 
+        /// <summary>
+        /// Attemps to unwrap the underlying type if the given type is <see cref="System.Nullable{T}"/>
+        /// </summary>
+        private static ITypeSymbol TryUnwrapNullableStruct(ITypeSymbol typeSymbol)
+        {
+            if (typeSymbol.TypeKind != TypeKind.Struct)
+                return typeSymbol;
+
+            var namedTypeSymbol = (INamedTypeSymbol)typeSymbol;
+            if (namedTypeSymbol.IsGenericType && namedTypeSymbol.ConstructedFrom.SpecialType == SpecialType.System_Nullable_T)
+            {
+                return namedTypeSymbol.TypeArguments[0];
+            }
+
+            return typeSymbol;
```

Comment:
>return typeSymbol; [](http://example.com/codeflow?start=12&length=18)

It looks like there are a couple of existing helper methods: IsNullableType() and GetNullableUnderlyingType(). Consider using those here.

Reply:
Yep these are exactly what I'm looking for.